CMakeLists.txt

```diff
@@ -0,0 +1,250 @@
+# SPDX-License-Identifier: Apache-2.0
+# Copyright 2013-2022 The Foundry Visionmongers Ltd
+
+#-----------------------------------------------------------------------
+# CMake preamble.
+
+# FindPython's Development.Module component was added in 3.18, and
+# pybind11 recommends 3.18.2+.
+cmake_minimum_required(VERSION 3.18.2)
+
+# Additional include directories for CMake utils.
+list(APPEND CMAKE_MODULE_PATH ${CMAKE_CURRENT_LIST_DIR}/cmake)
+
+if (APPLE AND NOT DEFINED CMAKE_OSX_DEPLOYMENT_TARGET)
+    # The value of this variable should be set prior to the first
+    # project() command invocation because it may influence
+    # configuration of the toolchain and flags.
+    set(CMAKE_OSX_DEPLOYMENT_TARGET "10.13" CACHE STRING "Minimum OS X deployment version")
+endif ()
+
+# Forbid in-source builds.
+set(CMAKE_DISABLE_IN_SOURCE_BUILD ON)  # NB: currently undocumented.
+# Forbid CMake from editing sources
+set(CMAKE_DISABLE_SOURCE_CHANGES ON)  # NB: currently undocumented.
+
+# By default, do not output "Up-to-date" messages during install. Can
+# be spammy when this project is included in another via CMake's
+# ExternalProject.
+if (NOT DEFINED CMAKE_INSTALL_MESSAGE)
+    set(CMAKE_INSTALL_MESSAGE LAZY)
+endif ()
+
+
+#-----------------------------------------------------------------------
+# Set a default build type if none was specified
+
+if (NOT CMAKE_BUILD_TYPE AND NOT CMAKE_CONFIGURATION_TYPES)
+    message(STATUS "Setting build type to 'Release' as none was specified.")
+    set(CMAKE_BUILD_TYPE Release CACHE STRING "Choose the type of build." FORCE)
+    # Set the possible values of build type for cmake-gui, ccmake
+    set_property(
+        CACHE CMAKE_BUILD_TYPE
+        PROPERTY STRINGS
+        "Debug"
+        "Release"
+        "MinSizeRel"
+        "RelWithDebInfo")
+endif ()
+
+
+#-----------------------------------------------------------------------
+# IDE helpers
+
+# Generate compile_commands.json to make it easier to work with clang
+# based tools. Note: `project(...)` invisibly defines
+# `CMAKE_EXPORT_COMPILE_COMMANDS` so this check must come before.
+if (NOT DEFINED CMAKE_EXPORT_COMPILE_COMMANDS)
+    set(CMAKE_EXPORT_COMPILE_COMMANDS ON)
+endif ()
+
+# Use folders to organise targets in IDEs via the FOLDER target
+# property.
+set_property(GLOBAL PROPERTY USE_FOLDERS ON)
+
+
+#-----------------------------------------------------------------------
+# Project metadata
+
+string(CONCAT project_description
```

Comment:
Is this just to fit into the 99 character column limit?

Reply:
Yes indeed. Frustrating that CMake has no neater way to split a single line string over multiple lines.

Reply:
This didn't work? https://cmake.org/cmake/help/latest/manual/cmake-language.7.html#quoted-argument

Reply:
The problem is that you then have to tackle indentation. However, escaping the newline works and on balance does look neater than having an entirely separate variable - updated in 80392ca7